### DISCUSSION TITLE
Should I unsubscribe when navigate?

### DISCUSSION THREAD

**Discussion by antoniogiroz** [urql-graphql/urql#2661] [Q&A] [answered]
Hi,

I'm using the `useSubscription` function in a Vue component (Vue 3 + Vite + TS). This component shows something is running, and my noob question is....

If navigate to other views and then return back to above view to see the progress, the useSubscription function is called since the component is mounted again. so

- Should I unsubscribe in `onUnmounted` hook?
- Could be this cause a possible memory leak?
- How can I know how many subscription are running or listen? Or maybe this does not make sense

Thanks a lot!

**Comment by JoviDeCroock** [comment, marked as answer]
Hey and welcome,

We should be unsubscribing for you when the `useSubscription` hook is invalidated, be this because of a variables change or the owning component unmounting. You can see all the active subscriptions at any time by opening up your browser devtools, navigating to network and click ws.

Unless my understanding fails me and Vue explicitly needs to be made aware of unmounting https://github.com/FormidableLabs/urql/blob/main/packages/vue-urql/src/useSubscription.ts#L144

EDIT: expected according to https://github.com/vuejs/composition-api/issues/293

**Reply by antoniogiroz** [reply]
Thanks @JoviDeCroock!
